### ISSUE TITLE
Panic when Dataframe is indexed with list of row indexes where at least one is out of bounds.

### ISSUE THREAD

**Issue by ghuls** [pola-rs/polars#1024]
#### Are you using Python or Rust?

**Python**

#### What version of polars are you using?
0.8.12

#### What operating system are you using polars on?

CentOS7

#### Describe your bug.

Panic when Dataframe is indexed with list of row indexes where at least one is out of bounds.

When just one row index (not as list) is given (which can be out of bounds), there is no crash.

#### What are the steps to reproduce the behavior?

```python
In [15]: df = pl.DataFrame(data=[pl.Series('col1', [3, 4, 8, 2], dtype=pl.Int64), pl.Series('col2', [6, 1, 5 , 4], dtype=pl.Int64)])

In [16]: df
Out[16]: 
shape: (4, 2)
╭──────┬──────╮
│ col1 ┆ col2 │
│ ---  ┆ ---  │
│ i64  ┆ i64  │
╞══════╪══════╡
│ 3    ┆ 6    │
├╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 4    ┆ 1    │
├╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 8    ┆ 5    │
├╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 2    ┆ 4    │
╰──────┴──────╯

In [17]: df[2]
Out[17]: 
shape: (1, 2)
╭──────┬──────╮
│ col1 ┆ col2 │
│ ---  ┆ ---  │
│ i64  ┆ i64  │
╞══════╪══════╡
│ 8    ┆ 5    │
╰──────┴──────╯

In [18]: df[[2, 1]]
Out[18]: 
shape: (2, 2)
╭──────┬──────╮
│ col1 ┆ col2 │
│ ---  ┆ ---  │
│ i64  ┆ i64  │
╞══════╪══════╡
│ 8    ┆ 5    │
├╌╌╌╌╌╌┼╌╌╌╌╌╌┤
│ 4    ┆ 1    │
╰──────┴──────╯

In [19]: df[[2, 4]]
thread '<unnamed>' panicked at 'index out of bounds: the len is 4 but the index is 4thread '', <unnamed>/software/polars/polars/polars-core/src/chunked_array/kernels/take.rs' panicked at ':index out of bounds: the len is 4 but the index is 4123', :/software/polars/polars/polars-core/src/chunked_array/kernels/take.rs20:
123note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
:20
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
<ipython-input-19-3f27d73e87b2> in <module>
----> 1 df[[2, 4]]

/software/miniconda3/envs/cisTopic/lib/python3.7/site-packages/polars/eager/frame.py in __getitem__(self, item)
    934                 else:
    935                     return wrap_df(
--> 936                         self._df.take([self._pos_idx(i, dim=0) for i in item])
    937                     )
    938             dtype = item.dtype

PanicException: index out of bounds: the len is 4 but the index is 4

In [20]: df[[4]]
thread '<unnamed>' panicked at 'index out of bounds: the len is 4 but the index is 4', /software/polars/polars/polars-core/src/chunked_array/kernels/take.rsthread '<unnamed>' panicked at 'index out of bounds: the len is 4 but the index is 4', /software/polars/polars/polars-core/src/chunked_array/kernels/take.rs::123123::2020

---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
<ipython-input-20-a7923f0dff33> in <module>
----> 1 df[[4]]

/software/miniconda3/envs/cisTopic/lib/python3.7/site-packages/polars/eager/frame.py in __getitem__(self, item)
    934                 else:
    935                     return wrap_df(
--> 936                         self._df.take([self._pos_idx(i, dim=0) for i in item])
    937                     )
    938             dtype = item.dtype

PanicException: index out of bounds: the len is 4 but the index is 4

In [21]: df[4]
Out[21]: 
shape: (0, 2)
╭──────┬──────╮
│ col1 ┆ col2 │
│ ---  ┆ ---  │
│ i64  ┆ i64  │
╞══════╪══════╡
╰──────┴──────╯

```

**Comment by ritchie46** [comment]
@jorgecarleitao this hits the arrow take kernels. I think that it would be better that the `safe` variants of the `take` kernels return an `Error` when indices are out of bounds, what do you think?

**Comment by ritchie46** [comment]
Fixed. On the just released `0.8.14` version:

```python
df = pl.DataFrame({
    'col1': [3, 4, 8, 2],
    'col2': [6, 1, 5 , 4]
})
df[[2, 4]]
```

```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-10-347d3e89b0e1> in <module>
      3     'col2': [6, 1, 5 , 4]
      4 })
----> 5 df[[2, 4]]

/opt/miniconda3/lib/python3.8/site-packages/polars/eager/frame.py in __getitem__(self, item)
    974                 else:
    975                     return wrap_df(
--> 976                         self._df.take([self._pos_idx(i, dim=0) for i in item])
    977                     )
    978             dtype = item.dtype

RuntimeError: Any(OutOfBounds("take indices are out of bounds"))
```
